### PR TITLE
add staller support to dsl tests

### PR DESCRIPTION
**Pull request by songgao**
This PR is pretty much #151 rebased to master, but created against master rather than `songgao/KBFS-1221`.

GitHub, please close #151 when this is merged.

### CODE REVIEW

**Comment by songgao** [comment]
How could it fail?

download

File: libkbfs/test_stallers.go [hunk 1]
```diff
@@ -38,6 +39,156 @@ const (
 	StallableMDPutUnmerged           StallableMDOp = "PutUnmerged"
 )
 
+const stallKeyStallEveryThing = ""
```
Comment (akalin-keybase) [inline libkbfs/test_stallers.go]: I think `Everything` instead of `EveryThing` is more idiomatic

File: libkbfs/test_stallers.go [hunk 2]
```diff
@@ -38,6 +39,156 @@ const (
...
+
+// NaïveStaller is used to stall certain ops in BlockOps or MDOps. Unlike
+// StallBlockOp and StallMDOp which provides a way to precisely control which
+// particular op is stalled by passing in ctx with corresponding stallKey,
+// NaïveStaller simply stalls all instances of specified op.
+type NaïveStaller struct {
+	config Config
+
+	mu             sync.RWMutex
+	blockOpsStalls map[StallableBlockOp]*naïveStallInfo
```
Comment (akalin-keybase) [inline libkbfs/test_stallers.go]: My rationale for preferring value types here is that returning pointers to lock-protected structures carry the danger that something might inadvertently modify them outside the lock.

But I don't think it matters too much in this case, so I'm okay with leaving it as pointers.

**Comment by akalin-keybase** [comment]
lgtm

File: libkbfs/test_stallers.go (hunk 1 above)
Comment (songgao) [inline libkbfs/test_stallers.go]: Oops; no idea why I put in the upper case T :D Thanks Fred

**Comment by songgao** [comment]
Thanks for the fast response Fred!

File: libkbfs/test_stallers.go (hunk 2 above)
Comment (songgao) [inline libkbfs/test_stallers.go]: Got you; I'll keep that in mind for future reference!

**Pull request merged**